doc/ir-semantics.text

```diff
@@ -38,11 +39,18 @@ arg ::=
 | e
 | &x
 
-e ::=    (simple) expressions
+e ::=     expression
+| se                    simple expression
+| primop(se*)           primitive operation (pure)
+| array(se)             empty array constructor
+| [se*]                 array constructor
+| x[se]                 array index
+| |x|                   array length
```

Comment:
I'd just use `length(array)`, but it's superficial. Otherwise the proposed primitives look good.